app/models/metadata_presenter/next_page.rb

```diff
@@ -11,17 +11,53 @@ def find
       else
         service.find_page_by_uuid(current_page_flow.default_next)
       end
+    ensure
+      session[:return_to_check_your_answer] = nil
     end
 
     private
 
     def check_answers_page
-      session[:return_to_check_your_answer] = nil
       service.pages.find { |page| page.type == 'page.checkanswers' }
     end
 
     def return_to_check_your_answer?
-      session[:return_to_check_your_answer].present?
+      session[:return_to_check_your_answer].present? &&
+        no_branch_ahead_for_current_page_and_user_did_not_changed_answers?
+    end
+
+    def no_branch_ahead_for_current_page_and_user_did_not_changed_answers?
+      no_branch_ahead_for_current_page? && user_did_not_changed_answers?
```

Comment:
```suggestion
      components_not_used_elsewhere? && answers_unchanged?
```

yeah difficult naming. it's just checking the current pages components aren't used in another branch later in the flow right?

Reply:
Yes, I wasn't happy with the naming hahaha. I like the suggestion for `answers_unchanged` but the components not used anywhere maybe don't reflect entirely what is happening. Maybe could be `components_not_used_for_branching?`. Wdyt?

Reply:
👍  yup, let's go with that :)